execution_engine_testing/tests/benches/auction_bench.rs

```diff
@@ -0,0 +1,297 @@
+        let transfer = ExecuteRequestBuilder::transfer(
+            *DEFAULT_ACCOUNT_ADDR,
+            runtime_args! {
+                    ARG_TARGET => *SYSTEM_ADDR,
+                    ARG_AMOUNT => MINIMUM_ACCOUNT_CREATION_BALANCE,
+                    ARG_ID => <Option<u64>>::None,
+            },
+        )
+        .build();
+        builder.exec(transfer);
+        builder.expect_success().commit();
+    }
+
+    for delegator_account in delegator_accounts {
+        let transfer = ExecuteRequestBuilder::transfer(
+            *DEFAULT_ACCOUNT_ADDR,
+            runtime_args! {
+                    ARG_TARGET => delegator_account,
+                    ARG_AMOUNT => DELEGATOR_INITIAL_BALANCE,
+                    ARG_ID => <Option<u64>>::None,
+            },
+        )
+        .build();
+        builder.exec(transfer);
+        builder.expect_success().commit();
+    }
+    builder
+}
+
+fn create_run_genesis_request(
+    validator_slots: u32,
+    genesis_accounts: Vec<GenesisAccount>,
+) -> RunGenesisRequest {
+    let exec_config = {
+        let wasm_config = *DEFAULT_WASM_CONFIG;
+        let system_config = *DEFAULT_SYSTEM_CONFIG;
+        let auction_delay = DEFAULT_AUCTION_DELAY;
+        let locked_funds_period_millis = DEFAULT_LOCKED_FUNDS_PERIOD_MILLIS;
+        let round_seigniorage_rate = DEFAULT_ROUND_SEIGNIORAGE_RATE;
+        let unbonding_delay = DEFAULT_UNBONDING_DELAY;
+        let genesis_timestamp_millis = DEFAULT_GENESIS_TIMESTAMP_MILLIS;
+        ExecConfig::new(
+            genesis_accounts,
+            wasm_config,
+            system_config,
+            validator_slots,
+            auction_delay,
+            locked_funds_period_millis,
+            round_seigniorage_rate,
+            unbonding_delay,
+            genesis_timestamp_millis,
+        )
+    };
+    RunGenesisRequest::new(
+        *DEFAULT_GENESIS_CONFIG_HASH,
+        *DEFAULT_PROTOCOL_VERSION,
+        exec_config,
+    )
+}
+
+fn setup_bench_run_auction(
+    group: &mut BenchmarkGroup<WallTime>,
+    validator_count: usize,
+    delegator_count: usize,
+) {
+    // Setup delegator public keys
+    let delegator_keys = generate_pks(delegator_count);
+    let validator_keys = generate_pks(validator_count);
+
+    let data_dir = TempDir::new().expect("should create temp dir");
+    let mut builder = run_genesis_and_create_initial_accounts(
+        data_dir.path(),
+        &validator_keys,
+        delegator_keys
+            .iter()
+            .cloned()
+            .map(|pk| pk.to_account_hash())
+            .collect::<Vec<_>>(),
+    );
+
+    let bids: Bids = builder.get_bids();
+    let active_bid = bids.get(&DEFAULT_ACCOUNT_PUBLIC_KEY.clone()).unwrap();
+    assert_eq!(
+        builder.get_purse_balance(*active_bid.bonding_purse()),
+        U512::from(VALIDATOR_BID_AMOUNT),
+    );
+    assert_eq!(
+        *active_bid.delegation_rate(),
+        1,
+        "unexpected delegation rate"
+    );
```

Comment:
Do we still need to keep this assertions? If so, can you add message to the first one about what's expected and why?

Reply:
No we're good, I'll remove them.